UKXattrMetadataStore.m

```diff
@@ -43,31 +29,25 @@ @implementation UKXattrMetadataStore
 //		If travLnk == YES, it follows symlinks.
 // -----------------------------------------------------------------------------
 
-+(NSArray*)		allKeysAtPath: (NSString*)path traverseLink:(BOOL)travLnk
++(NSArray*) allKeysAtPath: (NSString*)path traverseLink:(BOOL)travLnk
 {
-	NSMutableArray*	allKeys = [NSMutableArray array];
+	NSMutableArray<NSString*>*	allKeys = [NSMutableArray array];
 	size_t dataSize = listxattr( [path fileSystemRepresentation],
 								NULL, ULONG_MAX,
 								(travLnk ? 0 : XATTR_NOFOLLOW) );
 	if( dataSize == ULONG_MAX )
-		return allKeys;	// Empty list.
+		return nil;	// Empty list.
```

Comment:
Can't return NIL here, your header has ASSUME_NONNULL on. Also, why wouldn't you return an empty array here if it has no xattrs?